docs/book/src/topics/identities-use-cases.md

```diff
@@ -9,7 +9,7 @@ This is the identity used by the management cluster to provision infrastructure
    - Service Principal: A [service principal](https://docs.microsoft.com/en-us/azure/active-directory/develop/app-objects-and-service-principals) is an identity in AAD which is described by a TenantID, ClientID, and ClientSecret. The set of these three values will enable the holder to exchange the values for a JWT token to communicate with Azure. The values are normally stored in a file or environment variables.
    - Configuration:
       - Scope: Subscription
-      - Role: `Contributor` since the controller is responsible for creating resource groups and cluster resources within the group. To create a resource group within a subscription, one must have subscription contributor rights. Note, this role's scope can be reduced to Resource Group Contributor if all resource groups are created prior to cluster creation.
+      - Role: `Contributor` since the controller is responsible for creating resource groups and cluster resources within the group. To create a resource group within a subscription, one must have subscription contributor rights. Note, this role's scope can be reduced to Resource Group Contributor if all resource groups are created prior to cluster creation and the [VM image reference is specified](custom-images.md).  If no image is specified then Cluster Api for Azure quieres for reference image skus which requires subcription level access.
```

Comment:
Is this for e2e testing? Otherwise we're not currently hitting Azure APIs to select the image AFAIK? At least not until #2302 merges.

Reply:
> I don't see a way around requiring subscription-level access once #2302 merges, since we need to do the equivalent of `az vm image list`.

We should be aware of this. I think since these are reference images if a customer image is specified, we should avoid making the call so that customers don't need to give full subscription access to create clusters.

Reply:
Ok reproduced: 

create sp with
```
az ad sp create-for-rbac --name jjsclusterapitest
```
create a cluster with that sp:
```
az capi create -n win2 -l westus2 -w 
```


```
I0524 23:01:25.768179       1 azurecluster_controller.go:207] controller/azurecluster/controllers.AzureClusterReconciler.reconcileNormal "msg"="Reconciling AzureCluster" "name"="win2" "namespace"="default" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="AzureCluster" "x-ms-correlation-request-id"="<>"
E0524 23:01:25.908077       1 controller.go:317] controller/azurecluster "msg"="Reconciler error" "error"="failed to reconcile cluster services: failed to get availability zones: failed to get zones for location westus2: failed to refresh resource sku cache: could not list resource skus: 
compute.ResourceSkusClient#List: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code=\"AuthorizationFailed\"
 Message=\"The client '<>' with object id '<>' does not have authorization to perform action 'Microsoft.Compute/skus/read' over scope '/subscriptions/<>' or the scope is invalid. 
If access was recently granted, please refresh your credentials.\"" "name"="win2" "namespace"="default" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="AzureCluster"
```

I then added correct permissions:

```
az role assignment create --assignee <sp-objectid> --role Contributor --scope subscriptions/<sub>/resourceGroups/win2              
```

I waited a while but didn't get any new updates, I restarted the capz controller manager and things work.  So I guess this isn't an issue really an issue if things are set up properly.  

"If access was recently granted, please refresh your credentials"  seems to require a reboot of the capz controller manager

Reply:
as an update, the controller did refresh ~5 mins after the credentials where updated and things progressed